src/nvim/event/process.c

```diff
@@ -391,8 +391,13 @@ static void process_close_handles(void **argv)
 static void on_process_exit(Process *proc)
 {
   Loop *loop = proc->loop;
-  ILOG("exited: pid=%d status=%d stoptime=%" PRIu64, proc->pid, proc->status,
-       proc->stopped_time);
+  if (proc->term_signal) {
+    ILOG("exited: pid=%d signal=%d stoptime=%" PRIu64, proc->pid,
+         proc->term_signal, proc->stopped_time);
+  } else {
+    ILOG("exited: pid=%d status=%d stoptime=%" PRIu64, proc->pid,
+         proc->status, proc->stopped_time);
```

Comment:
If it's really necessary to have 2 fields then let's at least encapsulate in e.g. process_status() function.

Also does process_running() need update?

Reply:
I've changed it to use keep/use `status = 0` also for signals, but set `term_signal` then.

Reply:
`process_status()` might still make sense - although it is only used here then (AFAICS).